internal/terraform/node_resource_plan_instance.go

```diff
@@ -753,6 +753,12 @@ func (n *NodePlannableResourceInstance) importState(ctx EvalContext, addr addrs.
 		return nil, deferred, diags
 	}
 
+	// Providers are supposed to return an identity when importing by identity
+	if importTarget.Type().IsObjectType() && imported[0].Identity.IsNull() {
```

Comment:
👋🏻 I'm dipping my toe into Resource Identity stuff as this PR looked manageable for a newbie.

At this point could the importType still be either "ID" or "Identity"? If Terraform was trying to import by ID, would the identity being null still be a problem here?

Reply:
Basically I'm wondering if the condition also needs `importType == "Identity"` in it

Reply:
Do you think changing `importTarget.Type().IsObjectType()` to `importType == "Identity"` would make it more readable in this case?

The sprinkled `.IsObjectType()` checks trough the codebase feel a bit _odd_ to me

Reply:
I'm not sure what would be more descriptive if you're unfamiliar with the code. The prior knowledge of knowing the difference in usage of a string vs an object isn't far from knowing the difference between `ID` and `Identity`. I'd rather not have multiple fields that need to be updated in concert, since that invites them to end up out of sync, so the other option would probably be a struct containing an enum of the possible types, providing a type declaration with appropriate comments. We already have an `ImportTarget` type which could be used, though that might have been dropped because it makes some other part of the expansion code simpler.